src/charm.py

```diff
@@ -300,10 +299,13 @@ def _deploy_k8s_resources(self) -> None:
             raise ErrorWithStatus("K8S resources creation failed", BlockedStatus)
         self.model.unit.status = MaintenanceStatus("K8S resources created")
 
-    def _on_install(self, _):
-        """Perform installation only actions."""
+    def _on_pebble_ready(self, _):
+        """Log status of the container."""
+        if not self.container.can_connect():
+            self.logger.warning("Container {self._container_name} is not ready.")
+            return
```

Comment:
Does this need to be a stronger resolution?  If we get here and container.can_connect() fails, is it something that is eventually going to trigger another pebble_ready event that'll get this code firing again?  I'm not sure what happens if the container gets killed then comes up again

If this does not resolve itself, then we need an error here.  If it does resolve itself, we probably still need to set a status or something (as at this point, the charm isn't working)

Reply:
If assumption is pebble_ready_event == container is up, then yes, error should be raised instead.